gobblin-data-management/src/main/java/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -53,6 +71,10 @@ public DatasetCleaner(FileSystem fs, Properties props) throws IOException {
     } catch (InvocationTargetException exception) {
       throw new IOException(exception);
     }
+    this.executor =
+        Executors
+            .newFixedThreadPool(Integer.parseInt(props.getProperty(DATASET_CLEAN_THREAD_POOL_SIZE,
```

Comment:
I think it makes more sense to use a bounded variable size thread pool combined with a `RateBasedLimiter`.